packages/lesmis/src/components/ApproveButton.js

```diff
@@ -0,0 +1,78 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import CheckIcon from '@material-ui/icons/Check';
+import { IconButton } from '@tupaia/admin-panel/lib';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import MuiSnackbar from '@material-ui/core/Snackbar';
+import { SmallAlert } from '@tupaia/ui-components';
+import { useApproveSurveyResponse } from '../api/mutations/useApproveSurveyResponse';
+import { FlexCenter } from './Layout';
+import { GREEN } from '../constants';
+
+const Button = styled(IconButton)`
+  width: 56px;
+`;
+
+export const ApproveButton = ({ value: id }) => {
+  const [isApproved, setIsApproved] = useState(false);
+  const [showAlert, setShowAlert] = useState(false);
+  const { mutate, isLoading, isError, isSuccess } = useApproveSurveyResponse();
+
+  const handleClose = () => {
+    setShowAlert(false);
+  };
+
+  const handleClickAccept = async () => {
+    mutate(id, {
+      onSuccess: () => {
+        setShowAlert(true);
+        setIsApproved(true);
+      },
+      onError: () => {
+        setShowAlert(true);
+      },
+    });
+  };
+
+  return (
+    <>
+      {isApproved ? (
+        <FlexCenter>
+          <CheckIcon style={{ color: GREEN }} />
```

Comment:
Will the approved state ever show up in the page that has the approve button? I would have thought it might just disappear as the data table refreshes the data and that survey response no longer matches the filter, though that could be an uncomfortable experience for the user

Reply:
Yes good question. It does stay in the list until the page is refreshed or until you revisit the tab (no FE cache here). The approve button uses a lesmis query rather than dispatching a table action so the normal admin panel table refresh is not triggered. I did do a bit of playing around to see if I could get the refresh to work and it is possible in a couple of ways by either adding redux-connect to lesmis or by moving the ApproveButton component to AdminPanel and connecting to the store there. I thought that:
a) it's not great for maintenance as it adds lesmis concepts to admin panel or it adds redux to lesmis and
b) it wasn't necessarily the best user experience anyway

And so opted for some obvious visual feedback instead using the toast and green tick.